cmd/dlv/main.go

```diff
@@ -120,15 +124,28 @@ func main() {
 	})
 	go server.Run()
 
-	// Create and start a terminal
-	client := rest.NewClient(listener.Addr().String())
-	term := terminal.New(client)
-	err, status := term.Run()
+	status := 0
+	if !headless {
+		// Create and start a terminal
+		client := rest.NewClient(listener.Addr().String())
+		term := terminal.New(client)
+		err, status = term.Run()
+	} else {
+		ch := make(chan os.Signal)
+		signal.Notify(ch, sys.SIGINT)
+	runLoop:
+		for {
```

Comment:
Since we break immediately once we get a signal, is there need for the loop? Just block on the signal and stop the server.